backend/fms_core/services/sample.py

```diff
@@ -429,6 +429,124 @@ def pool_samples(process: Process,
+    listed in samples_info. These samples are not full samples, instead they have the information to create
+    their respective derived samples and biosamples. Meaning only one sample object is created: that of the pool.
+    This sample is then connected to the created derived samples and biosamples. This service is to be used by the
+    sample submission importer due to the necessity of creating pools without having to create futile sample objects
+    and containers.
+    The volume_ratio is calculated for each source sample and stored on the derivedbysample entry.
+
+
+    Args:
+        samples_info: a dict for source samples info.
+                      {"alias",
+                       "individual",
+                       "collection_site",
+                       "sample_kind",
+                       "tissue_source",
+                       "experimental_group",
+                       "library",
+                       "project",
+                       "volume"}
+
+        pool_name: the name given to the pool by the user.
+        container_destination: a container object that will receive the pool.
+        coordinates_destination: the coordinate on the container where the pool is stored.
+        reception_date: The date which the pool was received.
+        comment: Extra comments to attach to the pool.
+
+    Returns:
+        a tuple containing the following data.
+        sample_destination: a sample object that represents the new pool.
+        errors: errors that were encountered during the pool creation.
+        warnings: warnings to inform the user of potential mistakes or dangerous operations.
+    """
+    pool_sample_obj = None
+    errors = []
+    warnings = []
+
+    if not samples_info:
+        errors.append(f"Information must be provided about the samples to pool.")
+    if not container_destination:
+        errors.append(f"Destination container for extraction is required.")
+    if not pool_name:
+        errors.append(f"Pool Name is required.")
+    if not isinstance(reception_date, date):
+        errors.append(f"Reception date is not valid.")
+
+    if not errors:
+        # Calculate the total volume of the pool
+        pool_volume = sum(sample["volume"] for sample in samples_info)
+        for sample in samples_info:
+            sample_volume = sample["volume"]
+            # Calculate the volume ratio of each sample in the pool
+            sample["volume_ratio"] = decimal_rounded_to_precision(sample_volume / pool_volume)
+        # Create a a new sample - Concentration value is not set (need a QC to set it)
+        pool_data = dict(
+            name=pool_name,
+            volume=pool_volume,
+            creation_date=reception_date,
+            container=container_destination,
+            **(dict(comment=comment) if comment is not None else f"Automatically generated by pooling submission on {datetime.utcnow().isoformat()}Z"),
```

Comment:
```suggestion
            comment=comment if comment is not None else f"Automatically generated by pooling submission on {datetime.utcnow().isoformat()}Z"),
```
I don't understand why you would create a dict for the comment value and then unpack it. My linter complains that an unpacked dictionary is not a valid type of parameter for the dict() constructor.
Same thing for coordinates_destination.

Reply:
Is this a python thing that I don't get?

Reply:
Actually, only the comment line causes a problem, because if comment is none then you are trying to unpack a string, rather than an empty dict().

Reply:
It's not just the linter that complains...

![Screen Shot 2022-11-03 at 4 40 13 PM](https://user-images.githubusercontent.com/104078019/199829714-c96ba7cf-b512-462e-a1fd-8f3baf947b6b.png)

Reply:
The dict unpacking is an approach that comes from kwargs argument unpacking i think in order to not set some values that were not to be updated by leaving the value empty. We usually have empty dict for case where the parameter is None in order to skip the attribute. This is an inline solution that is more readable that an if else tree, but this is overzealous unpacking ;). we should not do a dict just to break it down.

Reply:
Ah, I see. It allows you to avoid adding keys with values of None to the dict. Instead, the keys are simply left out of the dict. I guess that helps if you want to call a function with the expanded dict as parameters, and the function defines default values for missing parameters. If a key is set to None then the function will use None for the value of the corresponding parameter, but if the key isn't present at all then the function will use any default value defined for the parameter.